l1-contracts/test/foundry/integration/deploy-scripts/script/Utils.sol

```diff
@@ -0,0 +1,231 @@
+
+import {AdminFacet} from "contracts/state-transition/chain-deps/facets/Admin.sol";
+import {ExecutorFacet} from "contracts/state-transition/chain-deps/facets/Executor.sol";
+import {GettersFacet} from "contracts/state-transition/chain-deps/facets/Getters.sol";
+import {MailboxFacet} from "contracts/state-transition/chain-deps/facets/Mailbox.sol";
+
+library Utils {
+    // Cheatcodes address, 0x7109709ECfa91a80626fF3989D68f67F5b1DD12D.
+    address internal constant VM_ADDRESS = address(uint160(uint256(keccak256("hevm cheat code"))));
+    // Create2Factory deterministic bytecode.
+    // https://github.com/Arachnid/deterministic-deployment-proxy
+    bytes internal constant CREATE2_FACTORY_BYTECODE =
+        hex"604580600e600039806000f350fe7fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffe03601600081602082378035828234f58015156039578182fd5b8082525050506014600cf3";
+
+    Vm internal constant vm = Vm(VM_ADDRESS);
+
+    /**
+     * @dev Get all selectors from the bytecode.
+     *
+     * Selectors are extracted by calling `cast selectors <bytecode>` from foundry.
+     * Then, the result is parsed to extract the selectors, removing
+     * the `getName()` selector if existing.
+     */
+    function getAllSelectors(bytes memory bytecode) internal returns (bytes4[] memory) {
+        string[] memory input = new string[](3);
+        input[0] = "cast";
+        input[1] = "selectors";
+        input[2] = vm.toString(bytecode);
+        bytes memory result = vm.ffi(input);
+        string memory stringResult = string(abi.encodePacked(result));
+
+        // Extract selectors from the result
+        string[] memory parts = vm.split(stringResult, "\n");
+        bytes4[] memory selectors = new bytes4[](parts.length);
+        for (uint256 i = 0; i < parts.length; i++) {
+            bytes memory part = bytes(parts[i]);
+            bytes memory extractedSelector = new bytes(10);
+            // Selector length 10 is 0x + 4 bytes
+            for (uint256 j = 0; j < 10; j++) {
+                extractedSelector[j] = part[j];
+            }
+            bytes4 selector = bytes4(vm.parseBytes(string(extractedSelector)));
+            selectors[i] = selector;
+        }
+
+        // Remove `getName()` selector if existing
+        bool hasGetName = false;
+        for (uint256 i = 0; i < selectors.length; i++) {
+            if (selectors[i] == bytes4(keccak256("getName()"))) {
+                selectors[i] = selectors[selectors.length - 1];
+                hasGetName = true;
+                break;
+            }
+        }
+        if (hasGetName) {
+            bytes4[] memory newSelectors = new bytes4[](selectors.length - 1);
+            for (uint256 i = 0; i < selectors.length - 1; i++) {
+                newSelectors[i] = selectors[i];
+            }
+            return newSelectors;
+        }
+
+        return selectors;
+    }
+
+    function getAdminSelectors() public pure returns (bytes4[] memory) {
```

Comment:
I think you can always use `getAllSelectors` over `getAdminSelectors `/`getExecutorSelectors`/`getGettersSelectors`/`getMailboxSelectors`

Reply:
We used it but this function requires ffi, which has to be set explicitly in test environment

Reply:
This approach uses command from foundry to inspect the contract, I don't see a way to call foundry command in testing environment without ffi.